sympy/series/tests/test_order.py

```diff
@@ -409,6 +409,16 @@ def test_order_subs_limits():
     assert Order(x**2).subs(x, y - 1) == Order((y - 1)**2, (y, 1))
     assert Order(10*x**2, (x, 2)).subs(x, y - 1) == Order(1, (y, 3))
 
+    # issue 19120
+    assert O(x).subs(x, O(x)) == O(x)
+    assert O(x**2).subs(x, x + O(x)) == O(x**2)
+    assert (x + O(x**2)).subs(x, x + O(x**2)) == x + O(x**2)
```

Comment:
Is this correct? Substituting would lead to O((x + O(x**2))**2) which would have an O(x**4) term.

Reply:
Yeah , I've made sure all results are consistent with master to check for any errors . See below for manual substitution of the same ! The second term `O(x**2)` captures the `O(x**4)` term as you pointed out .
```
>>> (x + O(x**2) + O((x + O(x**2))**2))
x + O(x**2)
```